[12.0][IMP] *rma*: Avoid RMAs of kits until rma_mrp is installed

ernestotejeda opened the pull request:
cc @Tecnativa TT27238

- Avoid creating RMAs from a portal Sales Order page for lines that don't have an associated picking (like Kit lines).
- Avoid creating RMAs from the website RMA for Kit products.

In PR https://github.com/OCA/rma/pull/183/files we are working on handling RMAs for kit products, but meanwhile, it is necessary to avoid creating RMAs from the website and portal for those kit products to avoid errors.

Commit:
[IMP] *rma*: Avoid RMAs of kits until rma_mrp is installed

Commit:
[IMP] *rma*: some improvements

pedrobaeza commented on website_rma/controllers/main.py:
```diff
@@ -29,6 +30,9 @@ def _get_website_rma_product_domain(self, q):
             ('name', '=ilike', "%{}%".format(q or '')),
             ("sale_ok", "=", True),
+            '|',
+            ('bom_ids.type', '!=', 'phantom'),
```
This is not valid, as you don't depend on `mrp` for doing this. You need to find another way.

chienandalu committed:
[FIX] rma_sale: product in multiple lines

If there are several stock moves with the same product in the picking we
won't be able to make the RMA

ernestotejeda commented:
@pedrobaeza changes done.
ping @chienandalu

chienandalu reviewed (commented):
Can you include this commit?: https://github.com/Tecnativa/rma/commit/244e9874ff337222467c92a34a792977703d67d1

pedrobaeza commented on website_rma/controllers/main.py:
```diff
@@ -26,10 +27,17 @@ def _get_website_rma_product_domain(self, q):
+        if "bom_ids" in request.env["product.product"]._fields:
+            domain += [
+                "|",
+                ("bom_ids.type", "!=", "phantom"),
```
This is still using things from `mrp`, so this can't be

pedrobaeza replied:
Not the ideal option, but finally there's no best one, so let's go with it.

pedrobaeza commented:
/ocabot merge minor

pedrobaeza commented:
Please fw-port it to v13
